lib/AST/NameLookup.cpp

```diff
@@ -2916,47 +2908,30 @@ directReferencesForDeclRefTypeRepr(Evaluator &evaluator, ASTContext &ctx,
                                    bool allowUsableFromInline) {
   DirectlyReferencedTypeDecls current;
 
-  auto *baseComp = repr->getBaseComponent();
-  if (auto *identBase = dyn_cast<IdentTypeRepr>(baseComp)) {
-    // If we already set a declaration, use it.
-    if (auto *typeDecl = identBase->getBoundDecl()) {
-      current = {1, typeDecl};
-    } else {
-      // For the base component, perform unqualified name lookup.
-      current = directReferencesForUnqualifiedTypeLookup(
-          identBase->getNameRef(), identBase->getLoc(), dc,
-          LookupOuterResults::Excluded, allowUsableFromInline);
-    }
-  } else {
-    current = directReferencesForTypeRepr(evaluator, ctx, baseComp, dc,
-                                          allowUsableFromInline);
-  }
-
-  auto *memberTR = dyn_cast<MemberTypeRepr>(repr);
-  if (!memberTR)
-    return current;
+  // If we already set a declaration, use it.
+  if (auto *typeDecl = repr->getBoundDecl()) {
+    current = {1, typeDecl};
 
-  // If we didn't find anything, fail now.
-  if (current.empty())
+    // NRVO.
```

Comment:
Do we get much benefit from doing this? Personally I would do the returns directly and eliminate the `current` variable, as in most cases there's only element and `TinyPtrVector` can store that inline (also I took a quick look at the assembly and it looks like this function gets inlined anyway). Additionally, assigning to `current` for the base decls of `MemberTypeRepr` seems weird since those aren't the in results we're going to return.

Reply:
Good point; Done!